website/server/controllers/top-level/payments/iap.js

```diff
@@ -54,12 +54,26 @@ api.iapAndroidVerify = {
       userId: user._id,
     });
 
-    await payments.buyGems({
-      user,
-      paymentMethod: 'IAP GooglePlay',
-      amount: 5.25,
-      headers: req.headers,
-    });
+    switch (purchaseData.productId) {
```

Comment:
Suggestion: Assign a value to `amount` in the switch, but make the API call afterwards to reduce redundancy.